monitoring/nodesstatus.go

```diff
@@ -94,3 +88,87 @@ func (r *nodesStatusChecker) Check(ctx context.Context, reporter health.Reporter
 		})
 	}
 }
+
+// NewNodeStatusChecker returns a Checker that validates availability
+// of a single kubernetes node
+func NewNodeStatusChecker(config KubeConfig, nodeName string) health.Checker {
+	nodeLister := kubeNodeLister{client: config.Client.CoreV1()}
+	return &nodeStatusChecker{
+		nodeLister: nodeLister,
+		nodeName:   nodeName,
+	}
+}
+
+// NewNodeStatusChecker returns a Checker that validates availability
+// of a single kubernetes node
+type nodeStatusChecker struct {
+	nodeLister
+	nodeName string
+}
+
+// Name returns the name of this checker
+func (r *nodeStatusChecker) Name() string { return NodeStatusCheckerID }
+
+// Check validates the status of kubernetes components
+func (r *nodeStatusChecker) Check(ctx context.Context, reporter health.Reporter) {
+	nodes, err := r.nodeLister.Nodes()
+	if err != nil {
+		reporter.Add(NewProbeFromErr(r.Name(), trace.UserMessage(err), trace.Wrap(err)))
+		return
+	}
+
+	var unavailableNode *v1.Node
+L:
+	for _, node := range nodes.Items {
```

Comment:
I'm curious why you decided to iterate over all nodes to find the local one instead of just querying it via filter parameter.

Reply:
This has been changed to use the filter.